### PR TITLE
Merge master to preview and move pnp helper to device only

### PR DESCRIPTION
**Pull request by prmathur-microsoft**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

**Comment by prmathur-microsoft** [comment]
\azp run

File: device/iot-device-samples/pnp-device-sample/pom.xml [hunk 1]
```diff
@@ -22,8 +22,14 @@
         <gson.version>2.8.6</gson.version>
         <slf4j-version>1.7.25</slf4j-version>
         <maven-jar-plugin-version>3.1.0</maven-jar-plugin-version>
+        <pnp-helper-version>1.0.0</pnp-helper-version>
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>com.microsoft.azure.sdk.iot.pnp</groupId>
+            <artifactId>pnp-helper</artifactId>
+            <version>${pnp-helper-version}</version>
```
Comment (vinagesh) [inline device/iot-device-samples/pnp-device-sample/pom.xml:31]: Is this defined somewhere? I see that it was removed from the root pom.

File: service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java [hunk 2]
```diff
@@ -61,7 +64,10 @@ private static void GetAndUpdateTwin() throws IOException, IotHubException {
         String propertyName = "targetTemperature";
         double propertyValue = 60.2;
         String componentName = "thermostat1";
-        twin.setDesiredProperties(PnpHelper.CreateComponentPropertyPatch(propertyName, propertyValue, componentName));
+        JsonObject patchJson = new JsonObject();
```
Comment (vinagesh) [inline service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java:67]: Can you add the comment we had in the helper which was showing how the patch json should look like?

File: service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java [hunk 3]
```diff
@@ -89,7 +95,7 @@ private static void InvokeMethodOnRootLevel() throws IOException, IotHubExceptio
     }
 
     private static void InvokeMethodOnComponent() throws IOException, IotHubException {
-        String methodToInvoke = PnpHelper.CreateComponentCommandName("thermostat1", "getMaxMinReport");
+        String methodToInvoke = "thermostat1" + "*" +  "getMaxMinReport";
```
Comment (vinagesh) [inline service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java:98]: Can you add the comments we had for this as well in the helper earlier?

File: device/iot-device-samples/pnp-device-sample/pom.xml (hunk 1 above)
Comment (prmathur-microsoft) [reply device/iot-device-samples/pnp-device-sample/pom.xml:31]: https://github.com/Azure/azure-iot-sdk-java/pull/866/files#diff-25a676cbb100784e8babd375ba346450R25

File: service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java (hunk 2 above)
Comment (prmathur-microsoft) [reply service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java:67]: coming in next commit

File: service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java (hunk 3 above)
Comment (prmathur-microsoft) [reply service/iot-service-samples/pnp-service-sample/temperature-controller-service-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/service/TemperatureController.java:98]: coming in next commit

**Pull request merged**